catalog/db_lineage_scanner_test.go

```diff
@@ -0,0 +1,201 @@
+package catalog
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/treeverse/lakefs/db"
+	"github.com/treeverse/lakefs/testutil"
+)
+
+func TestDBLineageScanner(t *testing.T) {
+	const numberOfObjects = 10
+
+	ctx := context.Background()
+	conn, uri := testutil.GetDB(t, databaseURI)
+	defer func() { _ = conn.Close() }()
+
+	c := TestCataloger{Cataloger: NewCataloger(conn), DbConnURI: uri}
+	baseBranchName := "b0"
+	repository := testCatalogerRepo(t, ctx, c, "repo", baseBranchName)
+
+	objSkip := []int{1, 2, 3, 5, 7, 11}
+	testSetupDBScannerData(t, ctx, c, repository, numberOfObjects, baseBranchName, objSkip)
+
+	bufferSizes := []int{1, 2, 8, 64, 512, 1024 * 4}
+
+	for _, bufSize := range bufferSizes {
+		_, _ = conn.Transact(func(tx db.Tx) (interface{}, error) {
+			// test lineage scanner
+			for branchNo := range objSkip {
+				branchName := "b" + strconv.Itoa(branchNo)
+				branchID := int64(branchNo + 1)
+				scanner := NewDBLineageScanner(tx, branchID, UncommittedID, &DBScannerOptions{BufferSize: bufSize})
+				for i := 0; scanner.Next(); i++ {
+					o := scanner.Value()
```

Comment:
IIUC we determine end-of-iteration by looking for `scanner.Value() == nil`, right?  So best test that it goes `nil` after `Next` returns `false`.

Reply:
We determine end-of-iteration by Next() returning false, so I'll test Value() must return non-nil and after iteration nil

Reply:
1. check the value returned from Next
2. added test to validate that when we have error we expect Value to be nil

Reply:
Yay! Thanks...